pkg/jit/jitdump.go

```diff
@@ -0,0 +1,386 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Specification:
+//   https://github.com/torvalds/linux/blob/master/tools/perf/Documentation/jitdump-specification.txt
+// Perf implementation:
+//   https://github.com/torvalds/linux/blob/master/tools/perf/util/jitdump.c
+//   https://github.com/torvalds/linux/blob/master/tools/perf/util/jitdump.h
+
+// Package jit provides a parser for Perf's JITDUMP files
+package jit
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+// JITHeader represent a jitdump file header.
+type JITHeader struct {
```

Comment:
nit: We should consider which types and fields we export if we don't depend on reflection or serialization.

Reply:
Agreed, but I am not sure how the package will be used yet, so I do not have an opinion on that, the ratio useful/expensive needs to be taken into account too